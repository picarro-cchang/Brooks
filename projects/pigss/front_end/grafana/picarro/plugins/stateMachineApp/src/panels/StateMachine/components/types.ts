export interface PlanFocus {
    row: number;
    column: number;
}

export interface BankConfig {
    chan_mask: number;
    clean: number;
}

export interface PlanStep {
    banks: { [key: number]: BankConfig };
    reference: number;
    duration: number;
}

export interface Plan {
    max_steps: number;
    panel_to_show: number;
    current_step: number;
    focus: PlanFocus;
    last_step: number;
    steps: { [key: string]: PlanStep };
    num_plan_files: number;
    plan_files: { [key: string]: string };
    plan_filename: string;
    bank_names: {
      [key: number]: {
        name: string;
        channels: { [key: number]: string };
      };
    };
}

export interface PlanLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    getPlanFileNames: () => void;
    loadedFileName: string;
}

export interface RunLayoutProps {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    layoutSwitch: () => void;
    fileNames: {}
    loadedFileName: string;
    getPlanFileNames: () => void;
}

export enum PanelTypes {
    NONE = 0,
    LOAD = 1,
    EDIT = 2,
    PREVIEW = 3
}

export enum PlanPanelTypes {
    PLAN = 0,
    LOAD = 1,
    SAVE = 2,
}

export interface BankPanelPlanOptions {
    bank: number;
    uistatus: {
      bank?: { [bankNum: string]: string };
      clean?: { [bankNum: string]: string };
      channel?: { [bankNum: string]: { [channelNum: string]: string } };
    };
    // ws_sender: (o: object) => void;
    plan: Plan;
    addChanneltoPlan: (bank: number, channel: number) => void;
}

export interface LoadPanelCommandOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
    planID: String;
  }

  export interface PreviewPanelOptions {
    plan: Plan;
    ws_sender: (o: object) => void;
    updatePanel: (x: number) => void;
    // loadPlan: (x: string) => void;
    // deleteFile: (x: string) => void;
    fileNames: {}
    cancelLoadPlan: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }


  export interface LoadPanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    getPlanFromFileName: (fileName: string) => void;
    deleteFile: (file: string) => void;
    loadedFileName: string;
  }

export interface PlanPanelOptions {
    uistatus: { [key: string]: any };
    plan: Plan;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    bankAddition: { [key: string]: number };
    fileName: string;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    planOverwrite: (s: object, x: String) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    updateSavedFileState: (plan: Plan) => void;
    planID: String;
  }

  export interface PlanSavePanelOptions {
    plan: Plan;
    // ws_sender: (o: object) => void;
    isEdited: boolean;
    updateFileName: (x: boolean) => void;
    planSaved: (f: string, d: object) => void;
    updatePanel: (x: number) => void;
    fileNames: {}
    deleteFile: (file: string) => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
  }

  export interface EditPanelOptions {
    uistatus: {
      [key: string]: any;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    planID: String;
  
  }
  export interface CommandPanelOptions {
    uistatus: {
      [key: string]: string;
    };
    ws_sender: (o: object) => void;
    plan: Plan;
    updatePanel: (x: number) => void;
    layoutSwitch: () => void;
    setModalInfo: (s: boolean, h: string, n: number, b: object, a: string) => void;
    loadedFileName: string;
    planID: String;
  }
