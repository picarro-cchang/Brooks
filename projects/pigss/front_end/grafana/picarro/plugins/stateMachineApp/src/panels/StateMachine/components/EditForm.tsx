import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div className="row">
        <label className="edit-label"> Bank {num}</label>
        {/* <input
          name={"bank" + num}
          className="col-sm-6 edit-input"
          type="text"
          value={this.props.plan.bank_names[num].name}
          onChange={event =>
            this.props.handleBankChange(
              event.target.value,
              num.toString()
            )
          }
          maxLength={14}
        /> */}
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={8}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
