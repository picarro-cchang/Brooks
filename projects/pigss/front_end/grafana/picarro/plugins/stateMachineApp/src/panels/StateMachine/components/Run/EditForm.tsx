import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";

class EditForm extends Component<any, any> {
  state = {
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  banks: any;
  bank_list: any;

  render() {
    const edit_list = [];
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const num in this.banks) {
      const value = this.banks[num];
      if (value === "READY") {
        this.bank_list.push(num);
        edit_list.push(<div key={"bank" + num} className="row">
            <label className="edit-label"> Bank {num}</label>
      </div>)
        for (let i = 1; i <= 8; i++) {
          let portNumber = (Number(num) - 1) * 8 + i
          edit_list.push(
            <div key={portNumber} className="row">
              <label className="edit-label"> Port {portNumber}: </label>
              <input                
                name={"bank" + num + i}
                className="col-sm-6 edit-input"
                type="text"
                onChange={event =>
                  this.props.handleChannelNameChange(
                    event.target.value,
                    num.toString(),
                    i
                  )
                }
                value={this.props.plan.bank_names[num].channels[i]}
                maxLength={15}
              />
            </div>
          );
        }
      }
    }
    return (
      <div className="panel-plan-inner">
        <div className="gf-form-group">
          {edit_list}
        </div>
      </div>);
  }
}

export default EditForm;
