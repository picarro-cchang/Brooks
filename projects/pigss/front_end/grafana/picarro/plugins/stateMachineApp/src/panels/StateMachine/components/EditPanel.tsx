import React, { Component, PureComponent, ReactText } from "react";
import { EditPanelOptions } from "../types";
import { EditForm } from "./EditForm";

class EditPanel extends PureComponent<EditPanelOptions> {
  state = {
    initialized: false,
    modal_info: {
      show: false,
      html: "",
      num_buttons: 0,
      buttons: {}
    },
    uistatus: {},
    plan: {
      bank_names: this.props.plan.bank_names
    }
  };
  constructor(props) {
    super(props);
  }

  banks: any;
  bank_list: any;

  handleSubmit = event => {
    event.preventDefault();
    let bankName, bankValue;
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (let key in this.banks) {
      let value = this.banks[key];
      if (value === "READY") {
        this.bank_list.push(key);
      }
    }
    const targets = event.target;
    const length = targets.length;
    for (let key in this.bank_list) {
      let bankNum = this.bank_list[key];
      bankName = "bank" + bankNum;
      bankValue = targets[bankName].value;
      const { bank_names } = { ...this.state.plan };
      const currentName = bank_names[bankNum];
      currentName.name = bankValue;
      this.setState({
        ...this.state.plan.bank_names,
        [bankNum]: { name: bankValue }
      });
      const { channels } = { ...this.state.plan.bank_names[bankNum] };
      const currentNames = channels;
      for (let i = 1; i < 9; i++) {
        const chanName = bankName + i;
        const chanValue = targets[chanName].value;
        currentNames[i] = chanValue;
        this.setState({
          ...this.state.plan.bank_names[bankNum].channels,
          [i]: chanValue
        });
      }
    }
    const channels1 = this.state.plan.bank_names[1].channels;
    const channels2 = this.state.plan.bank_names[2].channels;
    const channels3 = this.state.plan.bank_names[3].channels;
    const channels4 = this.state.plan.bank_names[4].channels;
    this.props.ws_sender({
      element: "edit_save",
      bank_names: {
        1: { name: this.state.plan.bank_names[1].name, channels: channels1 },
        2: { name: this.state.plan.bank_names[2].name, channels: channels2 },
        3: { name: this.state.plan.bank_names[3].name, channels: channels3 },
        4: { name: this.state.plan.bank_names[4].name, channels: channels4 }
      }
    });
  };
  render() {
    return (
      <div className="panel-edit">
        <h2 style={{ color: "#2f2f2f" }}>Edit Bank and Channel Names</h2>
        <form onSubmit={this.handleSubmit}>
          <EditForm
            uistatus={this.props.uistatus}
            plan={this.props.plan}
            ws_sender={this.props.ws_sender}
          />
          <div className="row text-center button-edit">
            <div className="col-sm-4">
              <button
                type="submit"
                className={"btn  btn-green btn-edit-panel btn-group-2"}
              >
                Ok
              </button>
            </div>
            <div className="col-sm-4">
              <button
                type="button"
                onClick={e => {
                  this.props.ws_sender({ element: "edit_cancel" });
                }}
                className={"btn btn-cancel btn-edit-panel btn-group-2"}
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
      </div>
    );
  }
}

export default EditPanel;
