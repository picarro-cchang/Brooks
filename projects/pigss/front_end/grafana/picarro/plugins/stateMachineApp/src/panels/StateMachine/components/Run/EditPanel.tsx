import React, { Component, PureComponent, ReactText } from "react";
import Modal from "react-responsive-modal";
import { EditPanelOptions, Plan } from "../types";
import EditForm from "./EditForm";

interface State {
  initialized: boolean,
  modal_info: {},
  uistatus: {},
  plan: {
    bank_names: {
      [key: string]: {
        name: string,
        channels: {
          [key: number]: string
        }
      }
    }
  },
  show: boolean,
  prevState: {}
}

class EditPanel extends PureComponent<EditPanelOptions, State> {
  constructor(props) {
    super(props);
    this.state = {
      initialized: false,
      modal_info: {
        show: false,
        html: "",
        num_buttons: 0,
        buttons: {}
      },
      uistatus: {},
      plan: {
        bank_names: this.props.plan.bank_names
      },
      show: false,
      prevState: {
        bank_names: {
          1: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: ""
            }
          },
          2: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: ""
            }
          },
          3: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: ""
            }
          },
          4: {
            name: "",
            channels: {
              1: "",
              2: "",
              3: "",
              4: "",
              5: "",
              6: "",
              7: "",
              8: ""
            }
          }
        }
      }
    };
    this.handleBankChange = this.handleBankChange.bind(this);
    this.handleChannelNameChange = this.handleChannelNameChange.bind(this);
  }

  

  banks: any;
  bank_list: any;

  componentDidMount = () => {
    // Store bank values in previous state
    this.setState({
      ...this.state,
      prevState: this.props.plan.bank_names
    });
  };

  handleBankChange = (value, num: string) => {
    this.setState({
      ...this.state,
      plan: {
        ...this.state.plan,
        bank_names: {
          ...this.state.plan.bank_names,
          [num]: {
            ...this.state.plan.bank_names[num],
            name: value
          }
        }
      }
    });
  };

  handleChannelNameChange(value, num: string, num2) {
    this.setState({
      ...this.state,
      plan: {
        ...this.state.plan,
        bank_names: {
          ...this.state.plan.bank_names,
          [num]: {
            ...this.state.plan.bank_names[num],
            channels: {
              ...this.state.plan.bank_names[num].channels,
              [num2]: value
            }
          }
        }
      }
    });
  }

  validateForm = bank_list => {
    let bankValue;
    for (const key in bank_list) {
      const bankNum = bank_list[key];
      bankValue = this.state.plan.bank_names[bankNum].name;
      if (bankValue.length < 1) {
        return false;
      } else if (bankValue.replace(/\s/g, "").length < 1) {
        return false;
      } else {
        for (let i = 1; i < 9; i++) {
          const chanValue = this.state.plan.bank_names[bankNum].channels[i];
          if (chanValue.length < 1) {
            return false;
          } else if (chanValue.replace(/\s/g, "").length < 1) {
            return false;
          }
        }
      }
    }
    return true;
  };

  handleSubmit = event => {
    event.preventDefault();
    this.bank_list = [];
    this.banks = this.props.uistatus.bank;
    for (const key in this.banks) {
      const value = this.banks[key];
      if (value === "READY") {
        this.bank_list.push(key);
      }
    }
    if (!this.validateForm(this.bank_list)) {
      this.setState({ show: true });
      return;
    }
    const channels1 = this.state.plan.bank_names[1].channels;
    const channels2 = this.state.plan.bank_names[2].channels;
    const channels3 = this.state.plan.bank_names[3].channels;
    const channels4 = this.state.plan.bank_names[4].channels;
    this.props.ws_sender({
      element: "edit_save",
      bank_names: {
        1: { name: this.state.plan.bank_names[1].name, channels: channels1 },
        2: { name: this.state.plan.bank_names[2].name, channels: channels2 },
        3: { name: this.state.plan.bank_names[3].name, channels: channels3 },
        4: { name: this.state.plan.bank_names[4].name, channels: channels4 }
      }
    });
  };

  handleClose = () => {
    this.setState({ show: false });
  };

  handleCancel = () => {
    // Set State to Previous Values
    this.setState({
      ...this.state,
      plan: {
        ...this.state.plan,
        bank_names: this.state.prevState
      }
    });
    this.props.ws_sender({ element: "edit_cancel" });
  };

  render() {
    const { show } = this.state;
    return (
      <div className="panel-edit">
        <h2 style={{ color: "#2f2f2f" }}>Edit Bank and Channel Names</h2>
        <form onSubmit={this.handleSubmit}>
          <EditForm
            handleBankChange={this.handleBankChange}
            handleChannelNameChange={this.handleChannelNameChange}
            uistatus={this.props.uistatus}
            plan={this.state.plan}
            ws_sender={this.props.ws_sender}
          />
          <div className="row text-center button-edit">
            <div className="col-sm-4">
              <button
                id={"submit-edit"}
                type="submit"
                className={"btn  btn-green btn-edit-panel btn-group-2"}
              >
                Ok
              </button>
            </div>
            <div className="col-sm-4">
              <button
                id={"cancel-btn"}
                type="button"
                onClick={this.handleCancel}
                className={"btn btn-cancel btn-edit-panel btn-group-2"}
              >
                Cancel
              </button>
            </div>
          </div>
        </form>
        <Modal open={show} onClose={this.handleClose} center>
          <h2 style={{ color: "black" }}>Uh Oh!</h2>
          <p style={{ color: "black" }}>
            Each Label must be at least one character in length.
          </p>
        </Modal>
      </div>
    );
  }
}

export default EditPanel;
