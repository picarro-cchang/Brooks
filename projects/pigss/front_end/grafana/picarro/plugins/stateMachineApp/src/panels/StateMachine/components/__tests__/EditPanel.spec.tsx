import React from 'react';
import {shallow, mount} from 'enzyme';
import Modal from 'react-responsive-modal';
import 'jest-styled-components'
import EditPanel from '../EditPanel';
import EditForm from '../EditForm';
import { EditPanelOptions } from '../../types';
import WS from 'jest-websocket-mock';
import validateForm from '../EditPanel';
 
const mockClick = jest.fn((element) => {return element});
const apiLoc = `${window.location.hostname}:8000/controller`;
const socketURL = `ws://${apiLoc}/ws`;

const defaultProps: EditPanelOptions = {
    uistatus: {
        "bank": {
            "1": "READY",
            "2": "DISABLED",
            "3": "READY",
            "4": "READY"
          },
    },
    plan: {
        max_steps: 32,
        panel_to_show: 3,
        current_step: 1,
        focus: {
            row: 1,
            column: 2
        },
        last_step: 1,
        steps: {
            "1": {
                "banks": {
                    "1": {
                    "clean": 0,
                    "chan_mask": 1
                    },
                    "2": {
                    "clean": 0,
                    "chan_mask": 0
                    }
                },
                "reference": 0,
                "duration": 30
            }
        },
        num_plan_files: 0,
        plan_filename: "",
        plan_files: {},
        bank_names: {
            "1": {
                "name": "B1",
                "channels": {
                    "1": "Channel 1",
                    "2": "Channel 2",
                    "3": "Ch. 3",
                    "4": "Ch. 4",
                    "5": "Ch. 5",
                    "6": "Ch. 6",
                    "7": "Ch. 7",
                    "8": "Ch. 8"
                }
            },
            "2": {
                "name": "B1",
                "channels": {
                    "1": "Channel 1",
                    "2": "Channel 2",
                    "3": "Ch. 3",
                    "4": "Ch. 4",
                    "5": "Ch. 5",
                    "6": "Ch. 6",
                    "7": "Ch. 7",
                    "8": "Ch. 8"
                }
            },
            "3": {
                "name": "Bank 2",
                "channels": {
                    "1": "Ch. 1",
                    "2": "Ch. 2",
                    "3": "Ch. 3",
                    "4": "Ch. 4",
                    "5": "Ch. 5",
                    "6": "Ch. 6",
                    "7": "Ch. 7",
                    "8": "Ch. 8"
                }
            },
            "4": {
                "name": "Bank 3",
                "channels": {
                    "1": "Ch. 1",
                    "2": "Ch. 2",
                    "3": "Ch. 3",
                    "4": "Ch. 4",
                    "5": "Ch. 5",
                    "6": "Ch. 6",
                    "7": "Ch. 7",
                    "8": "Ch. 8"
                }
            },
        }
    },
    ws_sender: mockClick,
};

describe('<EditPanel />', () => {
    const wrapper = shallow(<EditPanel {...defaultProps} />);
    const instance = wrapper.instance() as EditPanel;

    it('Match Snapshot', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('handleSubmit calls validation and websocket if validation', () => {
        wrapper.find('form').simulate('submit', {
            preventDefault: () => {
            }});
        expect(mockClick).toHaveBeenCalled();
        mockClick.mockClear();
    });

    it('Test Modal is open', () => {
        wrapper.setState({show: true});
        const modal = wrapper.find(Modal);
        expect(modal.props().open).toEqual(true);
        expect(wrapper).toMatchSnapshot();
    });

    it('Test handleClose', () => {
        wrapper.find(Modal).props().onClose()
        expect(wrapper.state('show')).toEqual(false);
    });

    it('Test Cancel', async () => {
        wrapper.setState({
            prevState: {
                1: {
                    name: "Bank 1",
                    channels: {
                        1: "Channel 1",
                        2: "Channel 2",
                        3: "Channel 3",
                        4: "Channel 4",
                        5: "Channel 5",
                        6: "Channel 6",
                        7: "Channel 7",
                        8: "Channel 8"
                    }
                },
                2: {
                    name: "Bank 2",
                    channels: {
                        1: "Channel 1",
                        2: "Channel 2",
                        3: "Channel 3",
                        4: "Channel 4",
                        5: "Channel 5",
                        6: "Channel 6",
                        7: "Channel 7",
                        8: "Channel 8"
                    }
                },
                3: {
                    name: "Bank 3",
                    channels: {
                        1: "Channel 1",
                        2: "Channel 2",
                        3: "Channel 3",
                        4: "Channel 4",
                        5: "Channel 5",
                        6: "Channel 6",
                        7: "Channel 7",
                        8: "Channel 8"
                    }
                },
                4: {
                    name: "Bank 4",
                    channels: {
                        1: "Channel 1",
                        2: "Channel 2",
                        3: "Channel 3",
                        4: "Channel 4",
                        5: "Channel 5",
                        6: "Channel 6",
                        7: "Channel 7",
                        8: "Channel 8"
                    }
                }
        }
        });
        const server = new WS(socketURL);
        const client = new WebSocket(socketURL);
        await server.connected;
        const cancel = wrapper.find('button#cancel-btn');
        cancel.simulate('click')
        const element = mockClick.mock.calls[0][0];
        client.send(element);
        expect(mockClick).toHaveBeenCalled();
        await expect(server).toReceiveMessage({element: "edit_cancel"})
        expect(server).toHaveReceivedMessages([{element: "edit_cancel"}])
        const test = wrapper.state('plan')['bank_names']
        expect(test).toEqual(wrapper.state('prevState'));
        mockClick.mockClear();
    });

    it('handleBankChange', () => {
        wrapper.find(EditForm).props().handleBankChange("Bank 1", "1");
        expect(wrapper.state()["plan"].bank_names["1"].name).toEqual("Bank 1");
    });

    it('handleChannelNameChange', () => {
        wrapper.find(EditForm).props().handleChannelNameChange("Channel Number 2", "1", 2);
        expect(wrapper.state()["plan"].bank_names["1"].channels[2]).toEqual("Channel Number 2");
    });

    it("Validate Form", () => {
        const component = new EditPanel(defaultProps);
        const boo = component.validateForm.call(defaultProps, [1, 3, 4]);
        expect(boo).toEqual(true);
    });
});

describe('<EditPanel /> Using Failing Bank Names', () => {
    const defaultPropsFail: EditPanelOptions = {
        uistatus: {
            "bank": {
                "1": "READY",
                "3": "READY",
                "4": "READY"
              },
        },
        plan: {
            max_steps: 32,
            panel_to_show: 3,
            current_step: 1,
            focus: {
                row: 1,
                column: 2
            },
            last_step: 1,
            steps: {
                "1": {
                    "banks": {
                        "1": {
                        "clean": 0,
                        "chan_mask": 1
                        },
                        "2": {
                        "clean": 0,
                        "chan_mask": 0
                        }
                    },
                    "reference": 0,
                    "duration": 30
                }
            },
            num_plan_files: 0,
            plan_filename: "",
            plan_files: {},
            bank_names: {
                "1": {
                    "name": "",
                    "channels": {
                        "1": "Channel 1",
                        "2": "Channel 2",
                        "3": "Ch. 3",
                        "4": "Ch. 4",
                        "5": "Ch. 5",
                        "6": "Ch. 6",
                        "7": "Ch. 7",
                        "8": "Ch. 8"
                    }
                },
                "2": {
                    "name": "   ",
                    "channels": {
                        "1": "Channel 1",
                        "2": "Channel 2",
                        "3": "Ch. 3",
                        "4": "Ch. 4",
                        "5": "Ch. 5",
                        "6": "Ch. 6",
                        "7": "Ch. 7",
                        "8": "Ch. 8"
                    }
                },
                "3": {
                    "name": "Bank 2",
                    "channels": {
                        "1": "   ",
                        "2": "Ch. 2",
                        "3": "Ch. 3",
                        "4": "Ch. 4",
                        "5": "Ch. 5",
                        "6": "Ch. 6",
                        "7": "Ch. 7",
                        "8": "Ch. 8"
                    }
                },
                "4": {
                    "name": "Bank 3",
                    "channels": {
                        "1": "",
                        "2": "Ch. 2",
                        "3": "Ch. 3",
                        "4": "Ch. 4",
                        "5": "Ch. 5",
                        "6": "Ch. 6",
                        "7": "Ch. 7",
                        "8": "Ch. 8"
                    }
                },
            }
        },
        ws_sender: mockClick,
    };
    const component = new EditPanel(defaultPropsFail);

    it("Validate Form", () => {
        const boo = component.validateForm.call(defaultPropsFail, [1]);
        const boo2 = component.validateForm.call(defaultPropsFail, [2]);
        const boo3 = component.validateForm.call(defaultPropsFail, [3]);
        const boo4 = component.validateForm.call(defaultPropsFail, [4]);
        expect(boo).toEqual(false);
        expect(boo2).toEqual(false);
        expect(boo3).toEqual(false);
        expect(boo4).toEqual(false);
    });

    it('handleSubmit calls validation and websocket if validation', () => {
        expect(component.handleSubmit.call(defaultPropsFail, {preventDefault: () => {}})).toBeUndefined();
    });
});
