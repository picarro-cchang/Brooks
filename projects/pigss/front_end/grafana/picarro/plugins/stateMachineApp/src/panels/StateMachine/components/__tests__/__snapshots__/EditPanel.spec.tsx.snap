// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditPanel /> Match Snapshot 1`] = `
<div
  className="panel-edit"
>
  <h2
    style={
      Object {
        "color": "#2f2f2f",
      }
    }
  >
    Edit Bank and Channel Names
  </h2>
  <form
    onSubmit={[Function]}
  >
    <EditForm
      handleBankChange={[Function]}
      handleChannelNameChange={[Function]}
      plan={
        Object {
          "bank_names": Object {
            "1": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 1",
            },
            "2": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 2",
            },
            "3": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 3",
            },
            "4": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 4",
            },
          },
        }
      }
      uistatus={
        Object {
          "bank": Object {
            "1": "READY",
            "2": "DISABLED",
            "3": "READY",
            "4": "READY",
          },
        }
      }
      ws_sender={[MockFunction]}
    />
    <div
      className="row text-center button-edit"
    >
      <div
        className="col-sm-4"
      >
        <button
          className="btn  btn-green btn-edit-panel btn-group-2"
          id="submit-edit"
          type="submit"
        >
          Ok
        </button>
      </div>
      <div
        className="col-sm-4"
      >
        <button
          className="btn btn-cancel btn-edit-panel btn-group-2"
          id="cancel-btn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={true}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "modalCenter": "styles_modalCenter__L9F2w",
        "overlay": "styles_overlay__CLSq-",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={true}
    onClose={[Function]}
    open={false}
    role="dialog"
    showCloseIcon={true}
    styles={Object {}}
  >
    <h2
      style={
        Object {
          "color": "black",
        }
      }
    >
      Uh Oh!
    </h2>
    <p
      style={
        Object {
          "color": "black",
        }
      }
    >
      Each Label must be at least one character in length.
    </p>
  </Modal>
</div>
`;

exports[`<EditPanel /> Test Modal is open 1`] = `
<div
  className="panel-edit"
>
  <h2
    style={
      Object {
        "color": "#2f2f2f",
      }
    }
  >
    Edit Bank and Channel Names
  </h2>
  <form
    onSubmit={[Function]}
  >
    <EditForm
      handleBankChange={[Function]}
      handleChannelNameChange={[Function]}
      plan={
        Object {
          "bank_names": Object {
            "1": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 1",
            },
            "2": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 2",
            },
            "3": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 3",
            },
            "4": Object {
              "channels": Object {
                "1": "Ch. 1",
                "2": "Ch. 2",
                "3": "Ch. 3",
                "4": "Ch. 4",
                "5": "Ch. 5",
                "6": "Ch. 6",
                "7": "Ch. 7",
                "8": "Ch. 8",
              },
              "name": "Bank 4",
            },
          },
        }
      }
      uistatus={
        Object {
          "bank": Object {
            "1": "READY",
            "2": "DISABLED",
            "3": "READY",
            "4": "READY",
          },
        }
      }
      ws_sender={[MockFunction]}
    />
    <div
      className="row text-center button-edit"
    >
      <div
        className="col-sm-4"
      >
        <button
          className="btn  btn-green btn-edit-panel btn-group-2"
          id="submit-edit"
          type="submit"
        >
          Ok
        </button>
      </div>
      <div
        className="col-sm-4"
      >
        <button
          className="btn btn-cancel btn-edit-panel btn-group-2"
          id="cancel-btn"
          onClick={[Function]}
          type="button"
        >
          Cancel
        </button>
      </div>
    </div>
  </form>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={true}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "modalCenter": "styles_modalCenter__L9F2w",
        "overlay": "styles_overlay__CLSq-",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    focusTrapOptions={Object {}}
    focusTrapped={true}
    onClose={[Function]}
    open={true}
    role="dialog"
    showCloseIcon={true}
    styles={Object {}}
  >
    <h2
      style={
        Object {
          "color": "black",
        }
      }
    >
      Uh Oh!
    </h2>
    <p
      style={
        Object {
          "color": "black",
        }
      }
    >
      Each Label must be at least one character in length.
    </p>
  </Modal>
</div>
`;
